### DISCUSSION TITLE
Problem with GitHub's CMake syntax highlighting for multi-line comments

### DISCUSSION THREAD

**Discussion by jamesETsmith** [community/community#60701] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

# Summary
I think there's a problem with GitHub's syntax highlighter for CMake when using multiline comments (`#[[ ... ]]`). It seems to register the multiline comment start (`#[[`) as a comment, but not anything after it.

# Details
Here's a minimal example:
```cmake
#[[
cmake_minimum_required(VERSION 3.20.2)

project(myproject)
]]
```

**Comment by RamtinKosari** [comment]
hi dear @jamesETsmith 
GitHub uses a syntax highlighter called Linguist, Linguist uses various techniques to determine the language of a code file, such as file extension, shebang line (e.g., #!/usr/bin/python), and patterns within the file's content. Once the language is identified, Linguist applies the appropriate syntax highlighting rules to colorize the code and improve readability.
but Linguist uses old version syntax system of cmake as far as i know,
i created an issue on related repo of linguist to see they fix it in future
until we receive update, you can use "#" for each line in cmake and make shour your code snippet is like this format :
```cmake
#[[
cmake_minimum_required(VERSION 3.20.2)

project(myproject)
]]
```

**Reply by RamtinKosari** [reply]
dont forget to use (```cmake) for init of cmake code snippet to determine your snippet is cmake

**Reply by jamesETsmith** [reply]
Just as a heads up, in the example above I did use ```cmake to start the code snippet.

**Comment by RamtinKosari** [comment]
Hi @jamesETsmith please check their response via this link :
https://github.com/github-linguist/linguist/issues/6496#event-9959728901

**Reply by jamesETsmith** [reply]
Thanks for raising this issue @RamtinKosari
